pkg/apis/azure/validation/infrastructure_test.go

```diff
@@ -289,6 +289,49 @@ var _ = Describe("InfrastructureConfig validation", func() {
 					"Detail": Equal("NatGateway is currently only supported for zoned cluster"),
 				}))
 			})
+
+			It("should return an error using a NatGateway as it not enabled but ip addresses are configured", func() {
```

Comment:
I'd also add a test for the ip address ranges here.

Reply:
yep done